Mathlib/Data/List/Sym.lean

```diff
@@ -0,0 +1,162 @@
+/-
```

Comment:
How easy is it to prove that the length is `Nat.multichoose`?

Reply:
Really easy (see `length_sym` at the end of this file).

I put a good amount of effort into trying to prove `List.Perm.sym`, but I might give up for now. That's the one key lemma to be able to define `Multiset.sym` and (since I have `List.Nodup.sym`) `Finset.sym`.

I think a way to prove it would be to use `sym_map` to switch to a list with `Nodup` and then carry out something like `symOptionSuccEquiv` for lists (use whether the head of the list is in the `Sym` element or not; if it is, erase to undo the `map` in the definition of `sym`).